core/repository/sparql/src/main/java/org/eclipse/rdf4j/repository/sparql/SPARQLConnection.java

```diff
@@ -195,10 +195,8 @@ public void setHttpClient(HttpClient httpClient) {
 
 	@Override
 	public void close() throws RepositoryException {
-		try {
+		try (client) {
```

Comment:
I've not seen try-with-resources used in this way before (autoclosing something that was defined out of scope of the actual try block). Could we do a quick smoke test that that actually works as expected?

Reply:
TBH I haven't tested all of them, and I'm thinking to roll back those (auto-generated) changes.
I assume it (mostly ?) works, but at the very least it is confusing...